### ISSUE TITLE
is feature extraction correct due to padding?

### ISSUE THREAD

**Issue by notes-ml** [agemagician/Ankh#24]
Thank you for this great protein LM. My question is regarding the feature extraction due to padding (given that the input sequences are from different length), in this case shouldn't the `attention_mask` be used as an additional input to the model for correct results, which makes sure that padding tokens of the inputs are ignored? like this:

```python

  model, tokenizer = ankh.load_large_model()
  model.eval()

  protein_sequences = ['MKALCLLLLPVLGLLVSSKTLCSMEEAINERIQEVAGSLIFRAISSIGLECQSVTSRGDLATCPRGFAVTGCTCGSACGSWDVRAETTCHCQCAGMDWTGARCCRVQPLEHHHHHH', 
  'GSHMSLFDFFKNKGSAATATDRLKLILAKERTLNLPYMEEMRKEIIAVIQKYTKSSDIHFKTLDSNQSVETIEVEIILPR']

  protein_sequences = [list(seq) for seq in protein_sequences]


  outputs = tokenizer.batch_encode_plus(protein_sequences, 
                                    add_special_tokens=True, 
                                    padding=True, 
                                    is_split_into_words=True, 
                                    return_tensors="pt")
  with torch.no_grad():
    embeddings = model(input_ids=outputs['input_ids'], attention_mask=outputs["attention_mask"])
```

thank you.

**Comment by hazemessamm** [comment]
Yes, that's correct. Thank you for pointing this out. I will fix this shortly. I will close this issue but feel free to re-open it if you want to add anything else.
